expressjs/app/api/safepaths/controller.js

```diff
@@ -15,35 +15,36 @@ const publicationFiles = require('../../lib/publicationFiles');
 exports.fetchSafePaths = async (req, res) => {
   const { params: { organization_id }, query: { type } } = req;
 
-  const organization = await organizations.findOne({ id: organization_id });
+  const organization = await organizations.fetchById(organization_id);
   if (organization) {
-    const record = await publications.findLastOne({ organization_id });
-    if (!record) {
+
+    // Find the most recent publication.
+    const publication = await publications.findLastPublicationAndCases({ organization_id, status: 'published' });
+    if (!publication) {
       return res.status(204).send('');
     }
 
-    let timeInterval = {
-      start_date: record.start_date.getTime() / 1000,
-      end_date: record.end_date.getTime() / 1000,
-    };
-
-    const trailsRecords = await trails.findInterval(timeInterval);
+    const trailsRecords = await trails.findInterval(publication);
     if (trailsRecords) {
       const intervalTrails = trails.getRedactedTrailFromRecord(trailsRecords);
       
       if (type === 'zip') {
-        let data = await publicationFiles.buildAndZip(organization, record, intervalTrails)
+        let data = await publicationFiles.buildAndZip(organization, publication, intervalTrails)
 
         res.status(200)
           .set({
             'Content-Type': 'application/octet-stream',
-            'Content-Disposition': `attachment; filename="${record.id}.zip"`,
+            'Content-Disposition': `attachment; filename="${publication.id}.zip"`,
             'Content-Length': data.length
           })
           .send(data)
+      } else if (type === 'json') {
+        let response = publicationFiles.build(organization, publication, intervalTrails)
+        res.status(200).json(response);
       }
       
-      let response = publicationFiles.build(organization, record, intervalTrails)
+      // TODO: Save to disk.
```

Comment:
What are we intending to save to disk in here?

Reply:
We will be.  That's next on my list.